examples/solvers/nuclear_norm_tomography.py

```diff
@@ -0,0 +1,122 @@
+
+The nuclear norm takes a vectorwise matrix norm, the spectral norm, i.e. the
+p-norm of the singular values. It introduces a coupling between the terms
+which allows better reconstructions by using the edge information from one term
+in reconstructing the other term.
+
+In this case we assume that ``g_2`` is much more noisy than ``g_1``, but we can
+still get an acceptable reconstruction.
+
+The data is assumed to be similar, but not exactly the same, for the channels.
+
+Note that this is an advanced example.
+"""
+
+
+import numpy as np
+import odl
+
+
+# --- Set up the forward operator (ray transform) --- #
+
+
+# Discrete reconstruction space: discretized functions on the rectangle
+# [-20, 20]^2 with 300 samples per dimension.
+space = odl.uniform_discr(
+    min_pt=[-20, -20], max_pt=[20, 20], shape=[100, 100], dtype='float32')
+
+# Make a parallel beam geometry with flat detector
+# Angles: uniformly spaced, n = 200, min = 0, max = 2 * pi
+angle_partition = odl.uniform_partition(0, 2 * np.pi, 300)
+# Detector: uniformly sampled, n = 200, min = -30, max = 30
+detector_partition = odl.uniform_partition(-30, 30, 300)
+geometry = odl.tomo.Parallel2dGeometry(angle_partition, detector_partition)
+
+# The implementation of the ray transform to use, options:
+# 'scikit'                    Requires scikit-image (can be installed by
+#                             running ``pip install scikit-image``).
+# 'astra_cpu', 'astra_cuda'   Requires astra tomography to be installed.
+#                             Astra is much faster than scikit. Webpage:
+#                             https://github.com/astra-toolbox/astra-toolbox
+impl = 'astra_cuda'
+
+# Create the forward operator, and also the vectorial forward operator.
+ray_trafo = odl.tomo.RayTransform(space, geometry, impl=impl)
+forward_op = odl.DiagonalOperator(ray_trafo, 2)
+
+# Create phantom, first contains only part of the information in the second
+# We do this by using a sub-set of the ellipses in the well known Shepp-Logan
+# phantom.
+ellipses = odl.phantom.shepp_logan_ellipses(space.ndim, modified=True)
+phantom = forward_op.domain.element(
+    [odl.phantom.ellipse_phantom(space, ellipses[:2]),
+     odl.phantom.ellipse_phantom(space, ellipses)])
+phantom.show('phantom')
+
+# Create phantom where second channel is highly noisy (SNR = 1)
+data = forward_op(phantom)
+data[1] += odl.phantom.white_noise(forward_op.range[1]) * np.mean(data[1])
+data.show('data')
+
+# Set up gradient and the vectorial gradient.
+gradient = odl.Gradient(ray_trafo.domain)
+pgradient = odl.DiagonalOperator(gradient, 2)
+
+# Create data discrepancy functionals
+l2err1 = odl.solvers.L2NormSquared(ray_trafo.range).translated(data[0])
+l2err2 = odl.solvers.L2NormSquared(ray_trafo.range).translated(data[1])
+
+# Scale the error term of the second channel so it is more heavily regularized.
+# Note that we need to use SeparableSum, otherwise the solver would not be able
+# to compute the proximal.
```

Comment:
Good!
Perhaps write out the expression in text, like `l2err([x, y]) = l2err1(x) + 0.1 * l2err(y)`, that would help understanding what's going on.